pkgs/by-name/li/libguestfs-with-appliance/package.nix

```diff
@@ -1,3 +1,12 @@
-{ libguestfs, libguestfs-appliance }:
+{
+  lib,
+  libguestfs,
+  libguestfs-appliance,
+}:
 
-libguestfs.override { appliance = libguestfs-appliance; }
+# https://github.com/NixOS/nixpkgs/issues/280881
+lib.warnIf (builtins.compareVersions libguestfs.version libguestfs-appliance.version > 0)
+  "libguestfs and libguestfs-appliance versions are out of sync, runtime errors may occur!"
```

Comment:
I think this check and warning message should be clarified a bit. Is it enough that one component is at least as new as the other, or should they have exactly the same version?

Reply:
This is just my attempt to prevent the same error creeping in again.
I also asked upstream for a clarification https://github.com/libguestfs/libguestfs/issues/156#issuecomment-2420335042.

Reply:
Ok.

In the meantime I think it would be good to update the message to something like `"libguestfs has a higher version than libguestfs-appliance (${libguestfs.version} > ${libguestfs-appliance.version}), runtime errors may occur!"`.

Reply:
No strict lockstep update required but `libguestfs` version should not be greater then the appliance.